Add time-logger extension

aramikuto opened the pull request:
## Description

This PR adds a new extension to log your work on the Google Calendar.

## Screencast

![time-logger-1](https://github.com/raycast/extensions/assets/116561995/c6dcd396-ba25-4a99-9f7c-4fc72b306a38)
![time-logger-2](https://github.com/raycast/extensions/assets/116561995/61fb8e0c-dcbb-4bea-bb1c-b58428593db4)
![time-logger-3](https://github.com/raycast/extensions/assets/116561995/c16ec3f7-839c-494e-9bec-ed14f258d1a5)


## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

pernielsentikaer commented on extensions/time-logger/src/gcalTimeLogger.tsx:
```diff
@@ -0,0 +1,127 @@
+                      <Action
+                        key="delete-epic"
+                        icon={Icon.XMarkCircle}
+                        title="Delete This Epic"
```
```suggestion
                        title="Delete This Epic"
                        style={Action.Style.Destructive}
```

pernielsentikaer replied:
You can also combine it with a [confirmAlert](https://developers.raycast.com/api-reference/feedback/alert#confirmalert)

pernielsentikaer commented on extensions/time-logger/src/gcalTimeLogger.tsx:
```diff
@@ -0,0 +1,127 @@
+                      />
+                    </>
+                  )}
+                  <Action
+                    key="create-epic-from-query"
+                    icon={Icon.PlusCircle}
+                    title="Create Epic From Search Query"
+                    shortcut={{ modifiers: ["cmd"], key: "n" }}
+                    onAction={createEpicFromQuery}
+                  />
```
Hide this when there is nothing in the search bar

pernielsentikaer commented on extensions/time-logger/src/gcalTimeLogger.tsx:
```diff
@@ -0,0 +1,127 @@
+                    shortcut={{ modifiers: ["cmd"], key: "n" }}
+                    onAction={createEpicFromQuery}
+                  />
+                  <Action
+                    key="edit-epic-description"
+                    icon={Icon.Redo}
+                    title="Edit Description"
+                    shortcut={{ modifiers: ["cmd"], key: "r" }}
+                    onAction={() => updateEpic(epic.name, inputEpicName)}
+                  />
```
What does this do? Nothing happens

pernielsentikaer commented on extensions/time-logger/src/gcalTimeLogger.tsx:
```diff
@@ -0,0 +1,127 @@
+                  ) : (
+                    <>
+                      <TimeLogAction setWorkingOnEpic={setWorkingOnEpicData} workingOnEpic={workingOnEpicData} />
+                      <Action
+                        key="discard-work"
+                        icon={Icon.DeleteDocument}
+                        title="Finish Work (Discard Work)"
+                        onAction={() => setWorkingOnEpicData(null)}
+                        shortcut={{ modifiers: ["cmd"], key: "d" }}
+                      />
```
Not sure what this does?

pernielsentikaer reviewed (changes requested):
How do you use this, could you make a screencast for a better understanding about why it's adding hours to a google calendar event 🙂

aramikuto commented:
@pernielsentikaer Thank you very much for the review. I have addressed the code review problems and reworked logic to make the interface more easy to understand.

I also added extension general information to the readme file:

> This extension is designed to help you log your work on specific epics in Google Calendar. It creates events with the activity name, start time, and end time.
> 
> 1. Register the names of the epics or activities you are working on. To add a new epic, press `⌘ + N`.
> 2. Before starting work on an activity you want to log in Google Calendar, press Enter on that activity in the list. The start time will appear on the right.
> 3. Once you're done with the work, press Enter on the activity you were working on. This action will open the Google Calendar event creation with necessary information filled in.

pernielsentikaer reviewed (approved):
Hi 👋 

Looks good to me, approved 🔥
